src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -16,6 +17,13 @@ public sealed partial class JsonSerializerOptions
     {
         internal const int BufferSizeDefault = 16 * 1024;
 
+        /// <summary>Tracks all live JsonSerializerOptions instances.</summary>
+        /// <remarks>Instances are added to the table in their constructor.</remarks>
+        internal static readonly ConditionalWeakTable<JsonSerializerOptions, object?> s_allOptionsInstances =
+            // TODO https://github.com/dotnet/runtime/issues/51159:
+            // Look into linking this away / disabling it when hot reload isn't in use.
```

Comment:
Can the linker be run on non-inbox TFMs? i.e. is it the case that non-net6.0 TFMs will always have this new code, even though we are only subscribing to hot reload notifications in the net6.0 TFM? If yes, do we care?

Reply:
Linker will have hard time to remove the field. It might be better to have the logic in nested static class

Reply:
> i.e. is it the case that non-net6.0 TFMs will always have this new code, even though we are only subscribing to hot reload notifications in the net6.0 TFM?

That will be the case with how I have it coded right now.  I can change it if you prefer.

> It might be better to have the logic in nested static class

Ok